src/OpenApi/test/Microsoft.AspNetCore.OpenApi.Tests/Integration/OpenApiDocumentConcurrentRequestTests.cs

```diff
@@ -0,0 +1,32 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net;
+using System.Net.Http;
+
+namespace Microsoft.AspNetCore.OpenApi.Tests.Integration;
+
+public class OpenApiDocumentConcurrentRequestTests(SampleAppFixture fixture) : IClassFixture<SampleAppFixture>
+{
+    [Fact]
+    public async Task MapOpenApi_HandlesConcurrentRequests()
+    {
+        // Arrange
+        var client = fixture.CreateClient();
+        var requests = new List<Task<HttpResponseMessage>>
+        {
+            client.GetAsync("/openapi/v1.json"),
+            client.GetAsync("/openapi/v1.json"),
+            client.GetAsync("/openapi/v1.json")
+        };
+
+        // Act
+        var results = await Task.WhenAll(requests);
```

Comment:
I wonder if using `Parallel` with a bigger number of operations would make this more likely to always run requests that race against each other?

Reply:
Thanks. That was an excellent idea. I switched to `Parallel.ForEachAsync` and it turned out that the `_operationTransformerContextCache` field must also be a `ConcurrentDictionary`.